wordpress-mobile/WordPress-Android
# [Reader Customization] Save reading preferences with device or app back buttons

> [!Warning]
> This should only be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/20506

~Add a "Done" button for saving Reader preferences instead of relying on the top app bar "back" button.~

After some discussion (p1711720310041669-slack-C06PGTY5PKK), it was decided to keep the dialog without a "Done" button and, instead, make sure both the system's back button and the top app bar back button save the preferences before exiting.

-----

## To Test:

Since the feature is still in development, follow the initial instructions in the "To Test" section of https://github.com/wordpress-mobile/WordPress-Android/pull/20506

- Go to Reader
- Open any post
- Open the Reading Preferences sheet (icon on top bar)
- Change the preferences
- Tap the system's back button
- **Verify** it saves the preferences accordingly
- Open the Reading Preferences sheet again
- Change the preferences
- Tap the back arrow on the top bar
- **Verify** it also saves the preferences

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A
    
2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [x] Portrait and landscape orientations. (there are bugs in landscape but not due to this PR, will be investigated and fixed on a different PR) 
- [ ] ~Talkback.~ (accessibility will be checked in a different PR)
